### ISSUE TITLE
Using custom_repository field causes error 

### ISSUE THREAD

**Issue by pwolthausen** [GoogleCloudPlatform/terraform-google-artifact-registry#39]
When using this module to create an artifact registry with a custom repository, the plan fails due to a conflict with both the custom_repository field and the public_repository field are both present.

## Expected Behavior
When a `custom_repository` field is provided, the artifact registry should be created. The `public_repository` field should be omitted

## Current Behavior
If the `custom_repository` field is provided, there is an error message during the terraform plan phase even if a null value is passed for this field.

```
│ Error: Conflicting configuration arguments
│ 
│   with module.artifact_registr["docker_k8s_gcr_io_proxy"].google_artifact_registry_repository.repo,
│   on .terraform/modules/artifact_registr/main.tf line 17, in resource "google_artifact_registry_repository" "repo":
│   17: resource "google_artifact_registry_repository" "repo" {
│ 
│ "remote_repository_config.0.docker_repository.0.custom_repository": conflicts with remote_repository_config.0.docker_repository.0.public_repository
╵
╷
│ Error: Conflicting configuration arguments
│ 
│   with module.artifact_registr["docker_k8s_gcr_io_proxy"].google_artifact_registry_repository.repo,
│   on .terraform/modules/artifact_registr/main.tf line 17, in resource "google_artifact_registry_repository" "repo":
│   17: resource "google_artifact_registry_repository" "repo" {
│ 
│ "remote_repository_config.0.docker_repository.0.public_repository": conflicts with remote_repository_config.0.docker_repository.0.custom_repository
```

## Possible Solution
Remove the `public_repository` field. This field only support the  value of DOCKER_HUB which it defaults to even if the field is not present (https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/artifact_registry_repository#nested_docker_repository)

**Comment by pwolthausen** [comment]
To reproduce the error, try the following resource

```
resource "google_artifact_registry_repository" "my-repo" {
  location      = "us-central1"
  repository_id = "example-docker-custom-remote"
  description   = "example remote custom docker repository with credentials"
  format        = "DOCKER"
  mode          = "REMOTE_REPOSITORY"
  remote_repository_config {
    description                 = "custom docker remote with credentials"
    disable_upstream_validation = true
    docker_repository {
      public_repository = "DOCKER_HUB"
      custom_repository {
        uri = "https://registry-1.docker.io"
      }
    }
  }
}
```

**Comment by pwolthausen** [comment]
this issue still persists and will resolve others

**Comment by prabhu34** [comment]
I see this has been updated in the APIs and I was able to create the repos without errors. Can you test again to confirm from your end? This should fix #46 as well.

API Reference: [Public Repo Enum](https://cloud.google.com/artifact-registry/docs/reference/rest/v1/projects.locations.repositories#publicrepository)
